1566 quota - release 1.11

hyi opened the pull request:
@pkdash @mjstealey I created another PR for quota against release 1.11 branch just for convenience in case quota needs to go in release 1.11. I kept the other corresponding PR against develop open for now since it has code reviews embedded. @pkdash Feel free to code review against either PR as you like, @mjstealey I'll leave it up to you which PR you will merge and close depending on whether quota implementation can go in to release 1.11 or not.

mjstealey commented:
@hyi - I think this would be fine to have in the RC as we can "mute" the quota pieces until given a final 👍 in production. Makes it easier to turn on if the pieces are already in place and not have to wait for a subsequent release.

hyi commented:
@mjstealey OK. I agree. "mute" the quota pieces are easy - just to comment out a few lines in a couple of places to "mute" the quota messages when users log in and on their profile page. The "mute" part is not done in this PR yet. Let me know if you want me to comment out those lines to "mute" them. I guess it all comes down to whether you are going to wire your script in for release 1.11 or release 1.12. This PR needs to go together with your script so that they work as expected. I can let you know which lines to comment out in order to "mute" quota messages, then you can do whatever needed for the next couple of releases. I think this is probably the easiest. Let me know which option you prefer.

mjstealey commented:
@hyi - I'd think if you create a `quota-info.md` file at the main level of the repository and document what needs to be done to turn quota **on** or **off** that would suffice.

It will also give me a place to document the backend script information for how everything gets wired together on the Jenkins / iRODS side of the fence.

hyi commented:
@mjstealey Have created ```quota-info.md``` in the quota branch and bring it up to date with release-candidate-1.11.0 branch. Feel free to edit ```quota-info.md``` as needed and let me know if anything I wrote there is not clear. I'll leave this PR open as is now and leave it to you to do whatever you think works best for the next couple of deployments. Do note I have already merged this PR to develop branch, so if you see any issues with that and need me to do anything, let me know.

mjstealey commented:
@pkdash - Could you do a final review?

I'd think getting this into 1.11 is a good thing with the understanding that we will not be enforcing quota at this time and would obscure the output to the user based on the [instructions](https://github.com/hydroshare/hydroshare/blob/1566-quota/quota-info.md) provided by @hyi.

We can then monitor usage and quota output behind the scenes transparently to the user and have some metrics to discuss prior to enabling quota enforcement in 1.12.

pkdash commented:
@mjstealey Will do another review of this PR today.

pkdash commented on quota-info.md:
````diff
@@ -0,0 +1,28 @@
+
+Although back-end code implementation for HydroShare quota management is included in the HydroShare code base and will make its way into next production deployment, front-end quota notification messaging and email notification languages may need some tweaks to better notify users about their quota situation and the actions they need to take. As a result, we need to turn quota front-end notificati ...
+
+- Comment out the line as shown in https://github.com/hydroshare/hydroshare/blob/develop/theme/templates/accounts/profile.html#L357 by enclosing the line ```<div class="text-muted">{{ quota_message }}</div>``` with ```{#``` and ```#}```.
````
Can't we turn on/off using settings.py?

hyi replied:
@pkdash @mjstealey I don't think it is a good idea to add a setting variable to turn on/off quota since this is only temporary workaround due to special quota implementation circumstances and impact. After quota is fully deployed, there will be no need to turn it off again.

pkdash commented on quota-info.md:
````diff
@@ -0,0 +1,28 @@
+# if add_msg:
+#    login_msg += add_msg
+```  
+- Comment out the lines as shown in https://github.com/hydroshare/hydroshare/blob/develop/theme/management/commands/update_used_storage.py#L85..L94 so they look like the following:
````
Same comment as above.

pkdash commented on hs_core/views/__init__.py:
```diff
@@ -117,7 +117,9 @@ def change_quota_holder(request, shortkey):
     except PermissionDenied:
```
Is there a unit test for this view function?

hyi replied:
@pkdash Yes, as shown here: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/tests/api/native/test_change_quota_holder.py

pkdash replied:
That unit test is not for the view function.

hyi replied:
@pkdash Can you point out an example for unit test for a view function for my reference to save me time to look for it myself just to speed things up? I will get it added first thing tomorrow morning so that this PR can make into release 1.11.

pkdash commented:
@hyi Are there unit tests to verify that user can't create resource with uploaded file or add a file to a resource if the user is over the quota limit?

hyi commented:
@pkdash Yes, there are unit tests to verify over-quota cases when creating new resources or adding files to existing resources. For example: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/tests/api/native/test_create_resource.py#L415

mjstealey commented:
@hyi - I would like to get this into 1.11.0, and if @pkdash is satisfied with the unit test question I'd think we can bring this in. I'll plan to rebuild prior to the programmers coordination call so that we can review quickly and then schedule the production deployment to www on Tuesday 7/18.

pkdash commented:
@hyi Once you have a unit test for the view function +1 from me.

hyi commented:
@pkdash I have updated test_change_quota_holder() api native unit tests to account for over-quota scenario, and also added unit test for the corresponding view function. See if you can +1 now.

pkdash commented:
+1
